Chore: 공통으로 사용되는 컴포넌트 위치 변경 및 네이밍 수정

jungjinbeom opened the pull request:
- 공통으로 사용되는 컴포넌트 위치 변경 및 네이밍 수정
- storage.ts 파일 추가
- Intro-> home 폴더, 파일, 컴포넌트 명 수정
- axios에서 지원해주는  AxiosRequestConfig 추가
- 상수에 as const  추가
- setStorage, getStorage 함수 명칭 오타 수정

jungjinbeom committed:
chore :리뷰 후 수정 및  컴포넌트 위치 변경
- 공통으로 사용되는 컴포넌트 위치 변경 및 네이밍 수정
- storage.ts 파일 추가
- Intro 폴더, 파일, 컴포넌트 명 수정
- AxiosRequestConfig 추가
- 상수에 as const  추가

junhyeong00 commented:
'리뷰 후 수정 및 컴포넌트 위치 변경' 이라는 PR보다는 세부사항인 '공통으로 사용되는 컴포넌트 위치 변경 및 네이밍 수정'이 PR단위가 되도록 작성해주시면 좋을 것 같습니다. (1작업 1PR) 그래야 최대한 충돌을 줄이고, 리뷰하기도 쉽고, 작업하기도 좋을 것 같습니다.

junhyeong00 commented:
본문에는 리뷰 시 참고하기 좋게 변경 사유도 함께 기재해주시면 좋을 것 같습니다.

Seung-wan commented on src/main.tsx:
```diff
@@ -2,7 +2,7 @@ import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import React from 'react';
 import ReactDOM from 'react-dom/client';
-import App from './App.tsx';
+import App from './app.tsx';
```
요기에 .tsx 확장자는 안붙어도 괜찮을 것 같네요~

Seung-wan commented on src/types/study-rooms.types.ts:
```diff
@@ -0,0 +1,16 @@
+export type PageKey = 'current' | 'total';
+
+export type Page = Record<PageKey, number>;
```
취향이지만 PageKey에 값이 얼마 없어서 이렇게 인라인해줄수도 있겠네요~
```suggestion
export type Page = Record<'current' | 'total', number>;
```

Seung-wan commented on src/pages/login/Login.tsx:
```diff
@@ -1,14 +1,13 @@
 import { login } from '@/apis/login/login';
 import { ROUTE_PATH } from '@/constant/routes';
-import { setLocalStorge } from '@/utils/util';
+import { setLocalStorge } from '@/utils/storage';
```
Storge -> Storage 오타인 것 같네요!

jungjinbeom committed:
chore: storage 함수 오타 수정
getLocalStorge -> getLocalStorage 함수 오타 수정
setLocalStorge -> setLocalStorage 함수 오타 수정
